demos/sample_azure_iot_pnp/sample_azure_iot_pnp.h

```diff
@@ -0,0 +1,104 @@
+/* Copyright (c) Microsoft Corporation.

```

Comment:
This is the main star of the PR, which defines the interface the samples will need to use to communicate with sample_azure_iot_pnp/sample_azure_iot_pnp_core.c (the common module that actually deals with the middleware functions).

Reply:
spitballing here. What if the `sample_azure_iot_pnp_core.c` was called `sample_azure_iot_pnp.c` and these interfaces were changed to `sample_azure_iot_pnp_device.c`. The main part of the app having a suffix and these not feels a little weird. 
So the heirarchy would go:

`main.c` -> `sample_azure_iot_pnp.c` -> `sample_azure_iot_pnp_device.c`.